lib/SPIRV/SPIRVInternal.h

```diff
@@ -191,6 +191,8 @@ enum SPIRAddressSpace {
   SPIRAS_Input,
   SPIRAS_Output,
   SPIRAS_Count,
+  SPIRAS_GlobalDevice = 11,
```

Comment:
I have a concern about `SPIRAS_GlobalDevice` value here: I'm not sure that this is reliable until we have this assignment documented/mentioned/used in llvm/llvm-project.

Someone might add new address space into upstream LLVM and use these values for some other purpose

Reply:
Fair enough

Reply:
Created https://reviews.llvm.org/D82174

Reply:
Taking these numeric values out of scope for now, could please review reset of the patch?

Reply:
@karolherbst
Hi, I see that you have added previously SPIRAS_Input and SPIRAS_Output into the map. Are they mapped to the actual address spaces in LLVM IR? If not I would love to shift them in the enum, claiming 5 and 6 to SPIRAS_GlobalDevice and SPIRAS_GlobalHost. The reason I want to do that is https://reviews.llvm.org/D82174 where I'm adding the appropriate address spaces in SPIR address space map. Currently the values assigned are 11 and 12, but it seems to be more logical to make a continues map without a gap from 4 to 11.
Thank you in advance.

Reply:
I am not sure about the actual llvm mapping there, I essentially just added them to fix some spirv-val issues with generated shaders and I think there needs to be some further changes as 1.4 or 1.5 changed the rules further.

But as long as the generated spirvs are still valid I really don't mind any changes in this area.

Reply:
Thanks! Looks like swapping has no affect (at least in local testing). Moved it here.